namesrv/src/main/java/org/apache/rocketmq/namesrv/processor/DefaultRequestProcessor.java

```diff
@@ -363,9 +364,16 @@ public RemotingCommand getRouteInfoByTopic(ChannelHandlerContext ctx,
                 topicRouteData.setOrderTopicConf(orderTopicConf);
             }
 
-            byte[] content = topicRouteData.encode(SerializerFeature.BrowserCompatible,
+            byte[] content;
+            Boolean standardJsonOly = requestHeader.getAcceptStandardJsonOnly();
+            if (request.getVersion() >= Version.V4_9_3.ordinal() || (null != standardJsonOly && standardJsonOly)) {
+                content = topicRouteData.encode(SerializerFeature.BrowserCompatible,
```

Comment:
The 4.9.3 has been released.

So here should > instead of >= ?

Reply:
Ahh... MQVersion#current should have been bumped up in the "develop" branch.

Reply:
Or avoid introducing version comparison?   Let clients decide to use `standardJsonOly` or not.

Reply:
@zhouxinyu Agrees with you.
It may be difficult to handle all the cases between different versions and different APIs. Just let the client decide.

Reply:
Generating invalid JSON is not something ignorable... It is a bug and a serious one, aliening this project from other communities. Anyone who ever writes non-Java code, communicating with RocketMQ,  would see the cumbersomeness.  It is OK to maintain compatibility with previously released clients and this is why this pull request and its predecessor pull request were created. However, this bug has to be fixed and eventually speak standard protocol. 

Back to the proposal of 'letting the client decide'. It seems reasonable at the first sight. But doing so means the future releases of the client would still require the name server continuously generate illegal JSON content. This still looks acceptable, until one day their infrastructure team would make transparent traffic management like envoy, linkerd or dapper. Boom! This issue turns up again.

Previously, I tried to add support for RocketMQ at envoy: https://github.com/envoyproxy/envoy/blob/main/contrib/rocketmq_proxy/filters/network/source/protocol.cc 
See how awkward the code is. I guess someone who read the code must laugh aloud, saying these guys do not understand JSON...amazing.

Reply:
@lizhanhui It is also OK to force the new clients to use standard JSON. I do not insist on it.

But we'd better do some extra effort to inform other client developers. For example, do notice on the mailing list and on the release notes.

Reply:
Also, it may be important to keep all the APIs to return standard JSON, not just some of them. This may be another effort.

Anyway a lot of Job.